website/docs/development/release.md

```diff
@@ -7,8 +7,8 @@ sidebar_label: Release process
 The release process may be automated in the future.
 
 - Checkout master
-- Update the Hydra version in `hydra/__init__.py`
-- Update NEWS.md with towncrier
-- Create a pip package for hydra-core: `python setup.py sdist bdist_wheel`
+- If releasing Hydra, update the version in `hydra/__init__.py`. For plugins, switch directory to `plugins/name-of-plugin` and update `setup.py`
+- Create draft release notes using `towncrier build --draft`. In case of plugins, add an extra flag for version i.e., `towncrier build --draft  --version x.y`
+- If the draft notes are correct, update NEWS.md using `towncrier build`. For plugins, use `towncrier build  --version x.y`
+- Create a pip package: `python setup.py sdist bdist_wheel`
```

Comment:
Please use two different sections, one for Hydra and one for plugins.

For both plugins and Hydra, we should call-out that the release should be done from the Hydra release branch and not from master.

Reply:
Done